react-accordion-component/my-app/src/Accordion.js

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from "react"
+import './Accordion.css'
+
+export default function Accordion({ topicsArray }) {
+  const [currentIndex, setCurrentIndex] = useState(null);
+
+  function HandleOnClick(index) {
+    index === currentIndex ? setCurrentIndex(null) : setCurrentIndex(index);
+  }
+
+  return (
+    <div className="container">
+      <Buttons data={topicsArray} current={currentIndex} onClick={HandleOnClick} />
+    </div>
+  );
+}
+
+function Buttons({ data, current, onClick }) {
```

Comment:
Are these really buttons? Aren' they topics? The fact they are implemented as buttons seems to be a detail the Accordion component shouldn't be concerned with.